### PR TITLE
Add ESLint, configure it and Prettier, add husky and lint-staged

### PR DESCRIPTION
**Pull request by sondr3**
This turned in to a much bigger PR than expected:

- I've added and configured ESLint with the configuration I use personally. I then had to fix all ~70 or so errors that I got in the various documents, mostly type errors from missing types on queries and so on. 
- I also ran Prettier on all the files, and added a `.prettierignore` so it doesn't format the files in `.cache` 
- I added `husky` and `lint-staged` so that you cannot commit files where either Prettier or ESLint have warnings/errors, so the code stays consistent across all contributors.
- I moved the design-documents out of the `src` directory because both ESLint and Prettier chocked when trying to figure out what kind of file it is.
- Finally, I added GitHub Actions that'll install, run `eslint src/**/*` and `prettier --check src` on pushes and PRs so that reviewers can see at a glance if the code is up to spec.

This should probably be merged after #58 because I expect there might be some merge conflicts.

### CODE REVIEW

**Review by andesyv — APPROVED** [review]
Wowies! Personally I like linting om commits, but I dislike the fact that it prevents you from commiting as I think that should be handled in the PRs instead.
Looks good to me, tho!

File: content/blog/notes/inf102.md [hunk 1]
```diff
@@ -100,18 +100,13 @@ in the wrong direction after a long time.
...
 
 Usually done recursively
 Divide and conquer.
-[2 , 8 , 5 , 3 , 9 , 4 , 1 , 7]
-[2 , 8 , 5 , 3] [9 , 4 , 1 , 7]
-[2 , 8] [5 , 3] [9 , 4] [1 , 7]
-[2 , 8] [3 , 5] [4 , 9] [1 , 7]
-[2 , 3 , 5 , 8] [1 , 4 , 7 , 9]
-[1 , 2 , 3 , 4 , 5 , 7 , 8 , 9]
+[2 , 8 , 5 , 3 , 9 , 4 , 1 , 7][2 , 8 , 5 , 3] [9 , 4 , 1 , 7][2 , 8] [5 , 3][9 , 4] [1 , 7][2 , 8] [3 , 5][4 , 9] [1 , 7][2 , 3 , 5 , 8] [1 , 4 , 7 , 9][1 , 2 , 3 , 4 , 5 , 7 , 8 , 9]
```
Comment (mrboen94) [inline content/blog/notes/inf102.md:109]: bilde
Well...

File: content/blog/notes/inf102.md [hunk 2]
```diff
@@ -848,24 +843,22 @@ Merge sort divides the array in to two equal sized parts (there may be a 1 size
 difference) and keeps doing this until it has only two elements left. It then
 sorts these two elements in the correct order.
 
-[2 , 5 , 3 , 7 , 8 , 9 , 8 , 2]
-[2 , 5 , 3 , 7] [8 , 9 , 8 , 2]
-[2 , 5] [3 , 7] [8 , 9] [8 , 2]
+[2 , 5 , 3 , 7 , 8 , 9 , 8 , 2][2 , 5 , 3 , 7] [8 , 9 , 8 , 2][2 , 5] [3 , 7][8 , 9] [8 , 2]
```
Comment (mrboen94) [inline content/blog/notes/inf102.md:846]: same issue

**Comment by sondr3** [comment]
Eh, given that a unformatted commit will immediately fail on push/PR there's no real benefit to not doing it. We should probably add a mention to the documentation that if you need help you can add `--no-verify` to a commit to ignore `husky`/`lint-staged` so that you can push it.

**Review by mrboen94 — CHANGES_REQUESTED** [review]

File: content/blog/notes/inf102.md (hunk 2 above)
Comment (sondr3) [reply content/blog/notes/inf102.md:846]: Tror TBH det dokumentet burde bare vært i PDF, det er voldsomt mye feil i konverteringen

Comment (mrboen94) [reply content/blog/notes/inf102.md:846]: Agreed

**Review by mrboen94 — APPROVED** [review]

**Pull request merged**